### ISSUE TITLE
[Bug]: Local instance profile view: Posts show none for "self"

### ISSUE THREAD

**Issue by Fmstrat** [LemmyNet/lemmy#4023]
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

When viewing posts for my user on the local instance (i.e. https://my.instance/u/user?page=1&sort=New&view=Posts) no posts are displayed.

When viewing my user elsewhere (i.e. https://another.instance/u/user@my.instance?page=1&sort=New&view=Posts), all (or most as federated) posts are displayed.

The count of posts above the post list are non-zero on both instances.

### Steps to Reproduce

1. View local user, see no posts
2. View remote user, see posts

### Technical Details

I tried to find this in the code itself, but I'm not familiar with the ORM and man it spits out some crazy queries. I did figure out where the problem is, though, **and would be happy (actually prefer) to make this my first contribution** if someone can point me in the right direction. When viewing `Posts` on the user page, this query is run:

```sql
SELECT
    "post"."id",
    "post"."name",
    "post"."url",
    "post"."body",
    "post"."creator_id",
    "post"."community_id",
    "post"."removed",
    "post"."locked",
    "post"."published",
    "post"."updated",
    "post"."deleted",
    "post"."nsfw",
    "post"."embed_title",
    "post"."embed_description",
    "post"."thumbnail_url",
    "post"."ap_id",
    "post"."local",
    "post"."embed_video_url",
    "post"."language_id",
    "post"."featured_community",
    "post"."featured_local",
    "person"."id",
    "person"."name",
    "person"."display_name",
    "person"."avatar",
    "person"."banned",
    "person"."published",
    "person"."updated",
    "person"."actor_id",
    "person"."bio",
    "person"."local",
    "person"."private_key",
    "person"."public_key",
    "person"."last_refreshed_at",
    "person"."banner",
    "person"."deleted",
    "person"."inbox_url",
    "person"."shared_inbox_url",
    "person"."matrix_user_id",
    "person"."admin",
    "person"."bot_account",
    "person"."ban_expires",
    "person"."instance_id",
    "community"."id",
    "community"."name",
    "community"."title",
    "community"."description",
    "community"."removed",
    "community"."published",
    "community"."updated",
    "community"."deleted",
    "community"."nsfw",
    "community"."actor_id",
    "community"."local",
    "community"."private_key",
    "community"."public_key",
    "community"."last_refreshed_at",
    "community"."icon",
    "community"."banner",
    "community"."followers_url",
    "community"."inbox_url",
    "community"."shared_inbox_url",
    "community"."hidden",
    "community"."posting_restricted_to_mods",
    "community"."instance_id",
    "community"."moderators_url",
    "community"."featured_url",
    "community_person_ban"."id",
    "community_person_ban"."community_id",
    "community_person_ban"."person_id",
    "community_person_ban"."published",
    "community_person_ban"."expires",
    "post_aggregates"."id",
    "post_aggregates"."post_id",
    "post_aggregates"."comments",
    "post_aggregates"."score",
    "post_aggregates"."upvotes",
    "post_aggregates"."downvotes",
    "post_aggregates"."published",
    "post_aggregates"."newest_comment_time_necro",
    "post_aggregates"."newest_comment_time",
    "post_aggregates"."featured_community",
    "post_aggregates"."featured_local",
    "post_aggregates"."hot_rank",
    "post_aggregates"."hot_rank_active",
    "post_aggregates"."community_id",
    "post_aggregates"."creator_id",
    "community_follower"."id",
    "community_follower"."community_id",
    "community_follower"."person_id",
    "community_follower"."published",
    "community_follower"."pending",
    "post_saved"."id",
    "post_saved"."post_id",
    "post_saved"."person_id",
    "post_saved"."published",
    "post_read"."id",
    "post_read"."post_id",
    "post_read"."person_id",
    "post_read"."published",
    "person_block"."id",
    "person_block"."person_id",
    "person_block"."target_id",
    "person_block"."published",
    "post_like"."score", coalesce(
    ("post_aggregates"."comments" - "person_post_aggregates"."read_comments"),
    "post_aggregates"."comments")
    FROM (
        (
            (
                (
                    (
                        (
                            (
                                (
                                    (
                                        (
                                            (
                                                ("post_aggregates"
                                                    INNER JOIN "person" ON ("post_aggregates"."creator_id" = "person"."id")
                                                )
                                                INNER JOIN "post" ON ("post_aggregates"."post_id" = "post"."id")
                                            )
                                            INNER JOIN "community" ON ("post_aggregates"."community_id" = "community"."id")
                                        )
                                        LEFT OUTER JOIN "community_person_ban" ON (
                                            ("post_aggregates"."community_id" = "community_person_ban"."community_id")
                                            AND ("community_person_ban"."person_id" = "post_aggregates"."creator_id")
                                        )
                                    ) LEFT OUTER JOIN "community_follower" ON (
                                        ("post_aggregates"."community_id" = "community_follower"."community_id")
                                        AND ("community_follower"."person_id" = '3')
                                    )
                                )
                                LEFT OUTER JOIN "post_saved" ON (
                                    ("post_aggregates"."post_id" = "post_saved"."post_id")
                                    AND ("post_saved"."person_id" = '3')
                                )
                            )
                            LEFT OUTER JOIN "post_read" ON (
                                ("post_aggregates"."post_id" = "post_read"."post_id")
                                AND ("post_read"."person_id" = '3')
                            )
                        )
                        LEFT OUTER JOIN "person_block" ON (
                            ("post_aggregates"."creator_id" = "person_block"."target_id")
                            AND ("person_block"."person_id" = '3')
                        )
                    ) LEFT OUTER JOIN "community_block" ON (
                        ("post_aggregates"."community_id" = "community_block"."community_id")
                        AND ("community_block"."person_id" = '3')
                    )
                ) LEFT OUTER JOIN "post_like" ON (
                    ("post_aggregates"."post_id" = "post_like"."post_id")
                    AND ("post_like"."person_id" = '3')
                )
            ) LEFT OUTER JOIN "person_post_aggregates" ON (
                ("post_aggregates"."post_id" = "person_post_aggregates"."post_id")
                AND ("person_post_aggregates"."person_id" = '3')
            )
        ) LEFT OUTER JOIN "local_user_language" ON (
            ("post"."language_id" = "local_user_language"."language_id")
            AND ("local_user_language"."local_user_id" = '2')
        )
    ) WHERE (
        (
            (
                (
                    ("post_aggregates"."creator_id" = '3')
                    AND ("post_read"."post_id" IS NULL)
                )
                AND ("local_user_language"."language_id" IS NOT NULL)
            )
            AND ("community_block"."person_id" IS NULL)
        )
        AND ("person_block"."person_id" IS NULL)
    )
    ORDER BY
        "post_aggregates"."featured_local" DESC ,
        "post_aggregates"."published" DESC 
    LIMIT '20' OFFSET '0'
```

This return 0 records. However, commenting:
```sql
-- AND ("post_read"."post_id" IS NULL)
```
allows all the posts to return. It seems Lemmy is expecting the post to not be read by the local current user for it to display.

### Version

0.18.4

**Comment by MV-GH** [comment]
Do you have have show read posts unchecked? Ifso that bug is known and I think is fixed #3691

**Comment by Fmstrat** [comment]
Doh, that's it. Apologies for the duplicate.
